sdk/ai/azure-ai-agents/samples/agents_tools/sample_agents_browser_automation.py

```diff
@@ -0,0 +1,130 @@
+            You can answer questions, provide information, and assist with various tasks 
+            related to web browsing using the Browser Automation tool available to you.
+            """,
+        tools=browser_automation.definitions,
+    )
+    # [END create_agent_with_browser_automation]
+
+    print(f"Created agent, ID: {agent.id}")
+
+    # Create thread for communication
+    thread = agents_client.threads.create()
+    print(f"Created thread, ID: {thread.id}")
+
+    # Create message to thread
+    message = agents_client.messages.create(
+        thread_id=thread.id,
+        role=MessageRole.USER,
+        content="""
+            Your goal is to report the percent of Microsoft year-to-date stock price change.
+            To do that, go to the website finance.yahoo.com.
+            At the top of the page, you will find a search bar.
+            Enter the value 'MSFT', to get information about the Microsoft stock price.
+            At the top of the resulting page you will see a default chart of Microsoft stock price.
+            Click on 'YTD' at the top of that chart, and report the percent value that shows up just below it.
+            """,
+    )
+    print(f"Created message, ID: {message.id}")
+
+    # Create and process agent run in thread with tools
+    print(f"Waiting for Agent run to complete. Please wait...")
+    run = agents_client.runs.create_and_process(thread_id=thread.id, agent_id=agent.id)
+    print(f"Run finished with status: {run.status}")
+
+    if run.status == "failed":
+        print(f"Run failed: {run.last_error}")
+
+    # Fetch run steps to get the details of the agent run
+    run_steps = agents_client.run_steps.list(thread_id=thread.id, run_id=run.id)
+    for step in run_steps:
+        print(f"Step {step.id} status: {step.status}")
+
+        if isinstance(step.step_details, RunStepToolCallDetails):
+            print("  Tool calls:")
+            tool_calls = step.step_details.tool_calls
+
+            for call in tool_calls:
+                print(f"    Tool call ID: {call.id}")
+                print(f"    Tool call type: {call.type}")
+
+                if isinstance(call, RunStepBrowserAutomationToolCall):
+                    print(f"    Browser automation input: {call.browser_automation.input}")
+                    print(f"    Browser automation output: {call.browser_automation.output}")
+
+                    print("    Steps:")
+                    for tool_step in call.browser_automation.steps:
+                        print(f"      Last step result: {tool_step.last_step_result}")
+                        print(f"      Current state: {tool_step.current_state}")
+                        print(f"      Next step: {tool_step.next_step}")
+                        print()  # add an extra newline between tool steps
+
+                print()  # add an extra newline between tool calls
+
+        print()  # add an extra newline between run steps
+
+    # Delete the agent when done
```

Comment:
For delete, would be good to inform user that deletion of the agent is optional and under the customers consideration. This is to avoid customer directly to use the pattern where they first create agent and at the end delete it.

        # Clean-up and delete the agent once the run is finished.
        # NOTE: Comment out this line if you plan to reuse the agent later.

Reply:
@jhakulin I updated the samples. Thanks!